### PR TITLE
win: name all public anonymous structs/unions

### PR DESCRIPTION
**Pull request by TomCrypto**
This addresses issue #205, by naming the anonymous structs and unions in `uv-win.h` to conform to the C89 spec and updating all the affected code. I tried to give meaningful and consistent names to the different private field groups, e.g.:
```c
union {
  struct { uv_stream_connection_fields };
  struct { uv_stream_server_fields };
};
```
becomes:
```c
union {
  struct { uv_stream_connection_fields } conn;
  struct { uv_stream_server_fields } serv;
} stream;
```
With similar naming for pipe, tcp, tty, fs and the overlapped I/O fields. Since it's a direct replacement it's a bit verbose, and it hits quite a lot of files since the private fields are used basically everywhere, I think it's nice that the fields are now all properly segmented and qualified since it indicates where they belong, though.

However I came across one issue and would like some comments: in `uv-common.c`, some functions assume the existence of an `nbufs` member directly inside the `uv_fs_t` struct, which is defined in `UV_FS_PRIVATE_FIELDS`. To me this doesn't seem compatible with the struct/union design currently being used in `uv-win.h` without anonymous structs/unions - should there be some internal functions to bridge the gap between the private field `nbufs` and its usage in the common libuv implementation?

----

What do you guys think, is this going in the right direction? Also, I'm intending to rebase and tidy up the comments and make a better git commit message, just looking for some feedback before I proceed since this is quite a large and intrusive commit for the windows implementation.

### CODE REVIEW

File: include/uv-win.h [hunk 1]
```diff
@@ -366,8 +366,8 @@ RB_HEAD(uv_timer_tree_s, uv_timer_s);
     struct {                                                                  \
       OVERLAPPED overlapped;                                                  \
       size_t queued_bytes;                                                    \
-    };                                                                        \
-  };                                                                          \
+    } io;                                                                     \
+  } req;                                                                      \
```
Comment (saghul) [inline include/uv-win.h]: Maybe call this one 'data'? Otherwise we end up with some "req.req.xyz" code, which looks like a mistake, but it's not.

**Comment by saghul** [comment]
Hi @TomCrypto, thanks a lot for this! I know it must have been tedious work...

The change mostly LGTM, with that naming nit I pointed out.

As for the change in uv-common, I think you solved it just the way I would have :-) It's an internal field, so we (libuv) are allowed to use it, and if we have to ifdef it for the time being, so be it. Please update the comment mentioning that there is a different structure layout.

/cc @piscisaureus and/or @bnoordhuis

**Comment by txdv** [comment]
That are a lot of changes.

**Comment by saghul** [comment]
@txdv but ABI stays kosher, so we should be fine :-)

File: include/uv-win.h (hunk 1 above)
Comment (TomCrypto) [inline include/uv-win.h]: @saghul Actually 'data' is already used by the `UV_REQ_FIELDS` as a void pointer, so it'd be a duplicate. I agree another name might be better though.

Comment (saghul) [inline include/uv-win.h]: @TomCrypto doh, you're right, I forgot about that :-S Let's see if someone else has some better name suggestions.

**Comment by TomCrypto** [comment]
@saghul Fixed the uv-common changes. And yes I am pretty sure the ABI is not changed, since anonymous structs/unions are basically just syntactic sugar afaik (not 100% sure though).

**Comment by txdv** [comment]
Makes stuff accessing clearer (immediately puts an emphasize on what part of the struct the variable belongs).

Looks good IMO.

File: src/win/pipe.c [hunk 2]
```diff
@@ -524,12 +524,12 @@ int uv_pipe_bind(uv_pipe_t* handle, const char* name) {
     goto error;
   }
 
-  if (uv_set_pipe_handle(loop, handle, handle->accept_reqs[0].pipeHandle, 0)) {
+  if (uv_set_pipe_handle(loop, handle, handle->pipe.serv.accept_reqs[0].pipeHandle, 0)) {
```
Comment (bnoordhuis) [inline src/win/pipe.c]: Style: long line, can you wrap this at 80 columns?

**Comment by bnoordhuis** [comment]
There are a couple of long lines but apart from that LGTM.  Nice work!

File: include/uv-win.h (hunk 1 above)
Comment (saghul) [inline include/uv-win.h]: @bnoordhuis @txdv do you guys have any suggestion for this one? Maybe "u"? (I think I've ssen u used somewhere, maybe not libuv)

Comment (bnoordhuis) [inline include/uv-win.h]: `u` is fine.

Comment (txdv) [inline include/uv-win.h]: What does 'u' stand for?

Comment (saghul) [inline include/uv-win.h]: "union", I've seen such convention here and there.

**Comment by TomCrypto** [comment]
Okay, I changed `req` to `u` as suggested, wrapped the long line, and squashed the commits.

**Comment by saghul** [comment]
@TomCrypto thanks a lot! Landed in :sparkles:  073323b :sparkles:

**Pull request closed without merging**